src/pages/studyView/charts/barChart/BarChart.tsx

```diff
@@ -256,6 +257,31 @@ export default class BarChart extends React.Component<IBarChartProps, {}>
     }
 
     public render() {
+        const containsNA = (element: BarDatum): boolean => {
```

Comment:
always better to define functions which are static like these outside of your control logic (in this case, render).